### ISSUE TITLE
For Gateway Load Balancer, enable_cross_zone_load_balancing is always false, regardless of what value is chosen

### ISSUE THREAD

**Issue by mabdelghany** [hashicorp/terraform-provider-aws#16311]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform CLI: v0.13.5
AWS Provider Version: v3.15.0_x5

### Affected Resource(s)

* aws_lb

### Terraform Configuration Files

```hcl
resource "aws_lb" "gwlb" {
  name                             = "${var.prefix}-gateway-lb"
  load_balancer_type               = "gateway"
  subnets                          = var.subnet_ids
  enable_cross_zone_load_balancing = true
}
```

Deploying this using `terraform apply` works fine. When you check the gateway load balancer in the console, you will see that the cross zone load balancing is disabled.

### Debug Output

I ran another `terraform plan` after my successful deployment and got this

```
resource "aws_lb" "gwlb" {
        arn                              = "arn:aws:elasticloadbalancing:us-east-1:xxxxxxxxxxxx:loadbalancer/gwy/xxxxxxxxxxxx-gateway-lb/xxxxxxxxxxxx"
        arn_suffix                       = "gwy/xxxxxxxxxxxx-gateway-lb/xxxxxxxxxxxx"
        drop_invalid_header_fields       = false
      ~ enable_cross_zone_load_balancing = false -> true
        enable_deletion_protection       = false
        enable_http2                     = true
        id                               = "arn:aws:elasticloadbalancing:us-east-1:xxxxxxxxxxxx:loadbalancer/gwy/xxxxxxxxxxxx-gateway-lb/xxxxxxxxxxxx"
        idle_timeout                     = 60
        internal                         = false
        load_balancer_type               = "gateway"
        name                             = "xxxxxxxxxxxx-gateway-lb"
        security_groups                  = []
        subnets                          = [
            "subnet-xxxxxxxxxxxx",
            "subnet-xxxxxxxxxxxx",
        ]
        tags                             = {
            "CreatedBy"   = "xxxxxxxxxxxx"
            "Environment" = "ProdNet"
            "Name"        = "xxxxxxxxxxxx-gateway-lb"
        }
        vpc_id                           = "vpc-xxxxxxxxxxxx"

        access_logs {
            enabled = false
        }

        subnet_mapping {
            subnet_id = "subnet-xxxxxxxxxxxx"
        }
        subnet_mapping {
            subnet_id = "subnet-xxxxxxxxxxxx"
        }
    }
```

### Expected Behavior

The cross zone load balancing should be enabled

### Actual Behavior

The cross zone load balancing is always disabled

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`

**Comment by bflad** [comment]
Fix submitted: https://github.com/hashicorp/terraform-provider-aws/pull/16314

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.17.0 of the Terraform AWS Provider, middle of next week. 👍
